.github/workflows/docker.yaml

```diff
@@ -0,0 +1,48 @@
+name: Docker
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+env:
+  VERSION: 0.2.1
+
+jobs:
+  docker:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l {0}
+    env:
+      DOCKER_BUILDKIT: 1
+      BUILDKIT_PROGRESS: plain
+      CLOUDSDK_CORE_DISABLE_PROMPTS: 1
+      BASE_IMAGE: australia-southeast1-docker.pkg.dev/cpg-common/images/cpg_aip
+    steps:
+    - uses: actions/checkout@main
+
+    - name: gcloud auth
+      uses: 'google-github-actions/auth@v0'
+      with:
+        credentials_json: ${{ secrets.GH_IMAGES_DEPLOYER_JSON }}
+
+    - name: set up gcloud sdk
+      uses: google-github-actions/setup-gcloud@v0
+      with:
+        project_id: cpg-common
+
+    - name: gcloud docker auth
+      run: |
+        gcloud auth configure-docker australia-southeast1-docker.pkg.dev
+    - name: build
+      run: |
+        docker build . -f Dockerfile --tag $BASE_IMAGE:${{ github.sha }}
```

Comment:
You could build this as whatever locally, not that it super matters anyway - but you're not restricted by the github.sha